main.py

```diff
@@ -448,6 +470,330 @@ def move_methods_workaround(dex_filename, dex_filename_last, in_dir, out_dir, de
+            contents.append("invoke-direct {v2, v3}, Lcom/android/server/pm/PackageManagerService;->allowSDWrite(Landroid/content/pm/PackageParser$Package;)V\n")
+            done_patching = True
+        else:
+            contents.append(old_contents[i])
+        i = i + 1
+    #
+    if not DEBUG_PROCESS:
+        if already_patched:
+            print_(" *** This file has been already patched... Exiting.")
+            exit_now(0)
+        elif not done_patching:
+            print_(os.linesep+"ERROR: The function to patch cannot be found, probably your version of Android is NOT supported.")
+            exit_now(89)
+        elif partially_patched:
+            print_(" *** Previous failed patch attempt, not including the allowsdwrite method again...")
+        else:
+            contents.extend(allowsdwrite)
+
+        f = open(to_patch, "w")
+        contents = "".join(contents)
+        f.write(contents)
+        f.close()
+    print_(" *** Patching succeeded.")
+    #
+    # Reassemble it
+    print_(" *** Reassembling classes...")
+    if not os.path.exists("out"):
+        os.makedirs("out/")
+    #
+    try:
+        assemble(smali_folder, "out/"+dex_filename, DEVICE_SDK, True)
+        if sys.platform == "win32":
+            subprocess.check_call(["attrib", "-a", "out/"+dex_filename])
+    except subprocess.CalledProcessError as e:  # ToDO: Check e.cmd
+        safe_file_delete(TMP_DIR+"/out/"+dex_filename)  # Remove incomplete file
+        output = e.output.decode("utf-8")
+        if e.returncode != 1 or "Unsigned short value out of range: 65536" not in output:
+            print_(os.linesep+output.strip())
+            print_(os.linesep+"Return code: "+str(e.returncode))
+            exit_now(83)
+        warning("The reassembling has failed (probably we have exceeded the 64K methods limit)")
+        warning("but do NOT worry, we will retry.", False)
+        move_methods_workaround(dex_filename, dex_filename_last, "services/", "out/", DEVICE_SDK)
+    #
+    # Backup the original file
+    BACKUP_FILE = os.path.join(OUTPUT_PATH, "services.jar.backup")
+    safe_copy(os.path.join(TMP_DIR, "services.jar"), BACKUP_FILE)
+    print_(os.linesep + "Your original services file is present at "+BACKUP_FILE)
+    if mode != 3:
+        print_(os.linesep + "If your device bootloop, please run this command on the pc when the connected device is inside recovery:" + os.linesep + "adb push \""+BACKUP_FILE+"\" /system/framework/services.jar")
+    #
+    # Put classes back in the archive
+    print_(" *** Recompressing services...")
+    compress(os.curdir+"/out/", "services.jar")
+    #
+    # Copy the patched file to the output folder
+    print_(" *** Copying the patched file to the output folder...")
+    safe_copy(os.path.join(TMP_DIR, "services.jar"), os.path.join(OUTPUT_PATH, "services.jar"))
+    #
+    # Clean up "framework.jar" disassembly folder.
+    #print_(" *** Deleting temporary directory:", TMP_DIR+"/smali-classes")
```

Comment:
__[flake8]__
* [E265] block comment should start with '# '

<!-- Comment by [SideCI](https://sideci.com) -->